src/mixins/ui/lasso-tool-ui.js

```diff
@@ -19,6 +19,92 @@ const MathExt = Draw.Math
 const DragEpsilon = 0.1 // this is a screen-space epsilon to validate that drag events are doing something
 // this can be relatively big since we're talking about screen coords
 
+/**
+ * Enum class that defines the different contexts in which a shape handler's destroy() method
+ * might be called. The enum context can be used in ShapeHandler-derived detroy() method to
+ * destroy the appropriate shapes that may have been created by the handler. For example,
+ * temporary shapes may be created while creating a new shape (i.e. a simple circle for each
+ * vertex for visual emphasis). The shape handler may want to only delete the temporary shapes
+ * in its destroy() method and not its primary shape.
+ */
+class DestroyType {
+  // NOTE: the below no-undef disabling is due to the outdated nature of
+  // the babel-eslint package and its apparent inappropriate handling of.
+  // static member variables
+
+  /**
+   * Signifies that the DestroyType is undefined.
+   * @enum
+   */
+  // eslint-disable-next-line no-undef
+  static kUndefined = new DestroyType(0, "unefined")
+
+  /**
+   * Signifies that destroy was called as a result of
+   * the shape handler successfully completing a shape
+   * @enum
+   */
+  // eslint-disable-next-line no-undef
+  static kCompleted = new DestroyType(1, "completed")
+
+  /**
+   * Signifies that destroy was called as a result of
+   * the shape handler creation being cancelled for some
+   * reason
+   * @enum
+   */
+  // eslint-disable-next-line no-undef
+  static kCancelled = new DestroyType(2, "cancelled")
+
+  /**
+   * Signifies that destroy was called as a result of
+   * the shape handler's parent tool control being
+   * deactivated
+   * @enum
+   */
+  // eslint-disable-next-line no-undef
+  static kDeactivated = new DestroyType(3, "deactivated")
+
+  /**
+   * Signifies that destroy was called as the tool being reset
+   * for some reason
+   * @enum
+   */
+  // eslint-disable-next-line no-undef
+  static kReset = new DestroyType(4, "reset")
+
+  /**
+   * Returns true if the destroy type is a cancellable type (i.e. cancelled or reset)
+   * @param {*} destroy_type
+   * @return {Boolean}
+   */
+  static isCancellableDestroyType(destroy_type) {
```

Comment:
utility used to classify whether the destroy type is ultimately a cancellable type, and therefore primary shapes need to be destroyed along with secondary shapes.